src/components/plots/barPlot.ts

```diff
@@ -237,38 +237,51 @@ export module Plots {
     }
 
     /**
-     * Gets the bar under the given pixel position (if [xValOrExtent]
-     * and [yValOrExtent] are {number}s), under a given line (if only one
-     * of [xValOrExtent] or [yValOrExtent] are {Extent}s) or are under a
-     * 2D area (if [xValOrExtent] and [yValOrExtent] are both {Extent}s).
-     *
-     * @param {number | Extent} xValOrExtent The pixel x position, or range of x values.
-     * @param {number | Extent} yValOrExtent The pixel y position, or range of y values.
-     * @returns {D3.Selection} The selected bar, or null if no bar was selected.
+     * Gets the {Plots.PlotData} that correspond to the given pixel position.
+     * 
+     * @param {Point} p The provided pixel position as a {Point}
+     * @return {Plots.PlotData} The plot data that corresponds to the {Point}.
      */
-    public getBars(xValOrExtent: number | Extent, yValOrExtent: number | Extent): D3.Selection {
-      if (!this._isSetup) {
-        return d3.select();
-      }
-
-      // currently, linear scan the bars. If inversion is implemented on non-numeric scales we might be able to do better.
-      var bars = this._datasetKeysInOrder.reduce((bars: any[], key: string) =>
-        bars.concat(this._getBarsFromDataset(key, xValOrExtent, yValOrExtent))
-      , []);
-
-      return d3.selectAll(bars);
+    public plotDataAt(p: Point): PlotData {
+      return this._getPlotData(p.x, p.y);
     }
 
-    private _getBarsFromDataset(key: string, xValOrExtent: number | Extent, yValOrExtent: number | Extent): any[] {
-      var bars: any[] = [];
+    /**
+     * Gets the {Plots.PlotData} that correspond to a given xRange/yRange
+     * 
+     */
+    public plotDataIn(bounds: Bounds): PlotData;
+    /**
+     * @param {Range} xRange The specified range of x values
+     * @param {Range} yRange The specified range of y values
+     * @return {Plots.PlotData} The plot data that corresponds to the ranges
+     */
+    public plotDataIn(xRange: Range, yRange: Range): PlotData;
+    public plotDataIn(xRangeOrBounds: Range | Bounds, yRange?: Range): PlotData {
+      var dataXRange = (<Range> xRangeOrBounds);
+      var dataYRange: Range = yRange;
+      if (yRange == null) {
+        var bounds = (<Bounds> xRangeOrBounds);
+        dataXRange = { min: bounds.topLeft.x, max: bounds.bottomRight.x };
+        dataYRange = { min: bounds.topLeft.y, max: bounds.bottomRight.y };
+      }
```

Comment:
Since it's consistent + it could save us in development downstream, let's have the
`if (yRange == null)` check at near the first line  I'm not really a fan of having the sanitation being done at the third line